src/Features/CSharp/Portable/GenerateMember/GenerateParameterizedMember/CSharpGenerateParameterizedMemberService.cs

```diff
@@ -172,8 +172,8 @@ protected override ImmutableArray<ITypeSymbol> DetermineTypeArguments(Cancellati
                 {
                     foreach (var typeArgument in ((GenericNameSyntax)State.SimpleNameOpt).TypeArgumentList.Arguments)
                     {
-                        var type = this.Document.SemanticModel.GetTypeInfo(typeArgument, cancellationToken).Type;
-                        result.Add(type);
+                        var typeInfo = this.Document.SemanticModel.GetTypeInfo(typeArgument, cancellationToken);
+                        result.Add(typeInfo.Type.WithNullability(typeInfo.Nullability.Annotation));
```

Comment:
so i see no way around this.  but this certainly seems like it would be repetitive to have to do everywhere (and it means auditing 100% of all usages of GetTypeInfo in the future :-/

While it doesn't help with the latter, this could maybe simplified by having extensions like `GetTypeInfoWithNullability`.

Reply:
Once we get comfortable with the pattern I think I'm going to have an analyzer that generally flags "you got an ITypeSymbol from a compiler API and didn't call WithNullability".

Reply:
And we *could* do extensions; the once nice advantage is if the compiler adopts this, all we have to do is just delete all the WithNullability() calls in a trivial update to the IDE, so I think having a consistent pattern (versus a bunch of special case extension methods) might be easier. But yes, this is still a bit clumsy.

Reply:
no worries.  i wasn't protesting.  just lamenting :)    I'm just heavily throwing my hat into the "oh please please please have the compiler expose this" camp.